src/App.tsx

```diff
@@ -1,61 +1,131 @@
 import './App.scss';
+import React, { useState } from 'react';
+import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import { User } from './types/User';
+import { Todo } from './types/Todo';
+import { TodoList } from './components/TodoList';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+function getUser(userId: number): User | null {
+  const foundUser = usersFromServer.find(user => user.id === userId);
+
+  return foundUser || null;
+}
+
+export const todos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUser(todo.userId),
+}));
+
+export const App: React.FC = () => {
+  const [visibleTodos, setVisibleTodos] = useState(todos);
+  const [title, setTitle] = useState('');
+  const [selectedUser, setSelectedUser] = useState<User | null>(null);
+  const [titleError, setTitleError] = useState(false);
+  const [userError, setUserError] = useState(false);
+
+  const maxId = Math.max(...todos.map(todo => todo.id)) + 1;
+
+  const handleChange = (
+    event: React.ChangeEvent<HTMLSelectElement | HTMLInputElement>,
+  ) => {
+    switch (event.target.id) {
+      case 'titleInput':
+        setTitleError(false);
+        setTitle(event.target.value);
+        break;
+
+      case 'userSelect':
+        setUserError(false);
+        setSelectedUser(getUser(Number(event.target.value)));
+        break;
+
+      default:
+        break;
+    }
+  };
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (!title) {
+      setTitleError(true);
+    }
+
+    if (!selectedUser) {
+      setUserError(true);
+    }
+
+    if (!title || !selectedUser) {
+      return;
+    }
+
+    const newTodo: Todo = {
+      id: maxId,
+      title,
+      userId: selectedUser ? selectedUser.id : -1,
```

Comment:
you already check it in the 59 line
![image](https://user-images.githubusercontent.com/91411692/235891218-e533253d-7f6d-45b6-be42-3652ce3a55ab.png)
```suggestion
      userId: selectedUser.id,
```